// @azure/ai-text-analytics/dist-esm/src/generated/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link StringIndexType} that the service accepts. */
export var KnownStringIndexType;
(function (KnownStringIndexType) {
    /** Returned offset and length values will correspond to TextElements (Graphemes and Grapheme clusters) confirming to the Unicode 8.0.0 standard. Use this option if your application is written in .Net Framework or .Net Core and you will be using StringInfo. */
    KnownStringIndexType["TextElementV8"] = "TextElement_v8";
    /** Returned offset and length values will correspond to Unicode code points. Use this option if your application is written in a language that support Unicode, for example Python. */
    KnownStringIndexType["UnicodeCodePoint"] = "UnicodeCodePoint";
    /** Returned offset and length values will correspond to UTF-16 code units. Use this option if your application is written in a language that support Unicode, for example Java, JavaScript. */
    KnownStringIndexType["Utf16CodeUnit"] = "Utf16CodeUnit";
})(KnownStringIndexType || (KnownStringIndexType = {}));
/** Known values of {@link PiiTaskParametersDomain} that the service accepts. */
export var KnownPiiTaskParametersDomain;
(function (KnownPiiTaskParametersDomain) {
    KnownPiiTaskParametersDomain["Phi"] = "phi";
    KnownPiiTaskParametersDomain["None"] = "none";
})(KnownPiiTaskParametersDomain || (KnownPiiTaskParametersDomain = {}));
/** Known values of {@link PiiCategory} that the service accepts. */
export var KnownPiiCategory;
(function (KnownPiiCategory) {
    KnownPiiCategory["ABARoutingNumber"] = "ABARoutingNumber";
    KnownPiiCategory["ARNationalIdentityNumber"] = "ARNationalIdentityNumber";
    KnownPiiCategory["AUBankAccountNumber"] = "AUBankAccountNumber";
    KnownPiiCategory["AUDriversLicenseNumber"] = "AUDriversLicenseNumber";
    KnownPiiCategory["AUMedicalAccountNumber"] = "AUMedicalAccountNumber";
    KnownPiiCategory["AUPassportNumber"] = "AUPassportNumber";
    KnownPiiCategory["AUTaxFileNumber"] = "AUTaxFileNumber";
    KnownPiiCategory["AUBusinessNumber"] = "AUBusinessNumber";
    KnownPiiCategory["AUCompanyNumber"] = "AUCompanyNumber";
    KnownPiiCategory["ATIdentityCard"] = "ATIdentityCard";
    KnownPiiCategory["ATTaxIdentificationNumber"] = "ATTaxIdentificationNumber";
    KnownPiiCategory["ATValueAddedTaxNumber"] = "ATValueAddedTaxNumber";
    KnownPiiCategory["AzureDocumentDBAuthKey"] = "AzureDocumentDBAuthKey";
    KnownPiiCategory["AzureIaasDatabaseConnectionAndSQLString"] = "AzureIAASDatabaseConnectionAndSQLString";
    KnownPiiCategory["AzureIoTConnectionString"] = "AzureIoTConnectionString";
    KnownPiiCategory["AzurePublishSettingPassword"] = "AzurePublishSettingPassword";
    KnownPiiCategory["AzureRedisCacheString"] = "AzureRedisCacheString";
    KnownPiiCategory["AzureSAS"] = "AzureSAS";
    KnownPiiCategory["AzureServiceBusString"] = "AzureServiceBusString";
    KnownPiiCategory["AzureStorageAccountKey"] = "AzureStorageAccountKey";
    KnownPiiCategory["AzureStorageAccountGeneric"] = "AzureStorageAccountGeneric";
    KnownPiiCategory["BENationalNumber"] = "BENationalNumber";
    KnownPiiCategory["BENationalNumberV2"] = "BENationalNumberV2";
    KnownPiiCategory["BEValueAddedTaxNumber"] = "BEValueAddedTaxNumber";
    KnownPiiCategory["BrcpfNumber"] = "BRCPFNumber";
    KnownPiiCategory["BRLegalEntityNumber"] = "BRLegalEntityNumber";
    KnownPiiCategory["BRNationalIdrg"] = "BRNationalIDRG";
    KnownPiiCategory["BGUniformCivilNumber"] = "BGUniformCivilNumber";
    KnownPiiCategory["CABankAccountNumber"] = "CABankAccountNumber";
    KnownPiiCategory["CADriversLicenseNumber"] = "CADriversLicenseNumber";
    KnownPiiCategory["CAHealthServiceNumber"] = "CAHealthServiceNumber";
    KnownPiiCategory["CAPassportNumber"] = "CAPassportNumber";
    KnownPiiCategory["CAPersonalHealthIdentification"] = "CAPersonalHealthIdentification";
    KnownPiiCategory["CASocialInsuranceNumber"] = "CASocialInsuranceNumber";
    KnownPiiCategory["CLIdentityCardNumber"] = "CLIdentityCardNumber";
    KnownPiiCategory["CNResidentIdentityCardNumber"] = "CNResidentIdentityCardNumber";
    KnownPiiCategory["CreditCardNumber"] = "CreditCardNumber";
    KnownPiiCategory["HRIdentityCardNumber"] = "HRIdentityCardNumber";
    KnownPiiCategory["HRNationalIDNumber"] = "HRNationalIDNumber";
    KnownPiiCategory["HRPersonalIdentificationNumber"] = "HRPersonalIdentificationNumber";
    KnownPiiCategory["HRPersonalIdentificationOIBNumberV2"] = "HRPersonalIdentificationOIBNumberV2";
    KnownPiiCategory["CYIdentityCard"] = "CYIdentityCard";
    KnownPiiCategory["CYTaxIdentificationNumber"] = "CYTaxIdentificationNumber";
    KnownPiiCategory["CZPersonalIdentityNumber"] = "CZPersonalIdentityNumber";
    KnownPiiCategory["CZPersonalIdentityV2"] = "CZPersonalIdentityV2";
    KnownPiiCategory["DKPersonalIdentificationNumber"] = "DKPersonalIdentificationNumber";
    KnownPiiCategory["DKPersonalIdentificationV2"] = "DKPersonalIdentificationV2";
    KnownPiiCategory["DrugEnforcementAgencyNumber"] = "DrugEnforcementAgencyNumber";
    KnownPiiCategory["EEPersonalIdentificationCode"] = "EEPersonalIdentificationCode";
    KnownPiiCategory["EUDebitCardNumber"] = "EUDebitCardNumber";
    KnownPiiCategory["EUDriversLicenseNumber"] = "EUDriversLicenseNumber";
    KnownPiiCategory["EugpsCoordinates"] = "EUGPSCoordinates";
    KnownPiiCategory["EUNationalIdentificationNumber"] = "EUNationalIdentificationNumber";
    KnownPiiCategory["EUPassportNumber"] = "EUPassportNumber";
    KnownPiiCategory["EUSocialSecurityNumber"] = "EUSocialSecurityNumber";
    KnownPiiCategory["EUTaxIdentificationNumber"] = "EUTaxIdentificationNumber";
    KnownPiiCategory["FIEuropeanHealthNumber"] = "FIEuropeanHealthNumber";
    KnownPiiCategory["FINationalID"] = "FINationalID";
    KnownPiiCategory["FINationalIDV2"] = "FINationalIDV2";
    KnownPiiCategory["FIPassportNumber"] = "FIPassportNumber";
    KnownPiiCategory["FRDriversLicenseNumber"] = "FRDriversLicenseNumber";
    KnownPiiCategory["FRHealthInsuranceNumber"] = "FRHealthInsuranceNumber";
    KnownPiiCategory["FRNationalID"] = "FRNationalID";
    KnownPiiCategory["FRPassportNumber"] = "FRPassportNumber";
    KnownPiiCategory["FRSocialSecurityNumber"] = "FRSocialSecurityNumber";
    KnownPiiCategory["FRTaxIdentificationNumber"] = "FRTaxIdentificationNumber";
    KnownPiiCategory["FRValueAddedTaxNumber"] = "FRValueAddedTaxNumber";
    KnownPiiCategory["DEDriversLicenseNumber"] = "DEDriversLicenseNumber";
    KnownPiiCategory["DEPassportNumber"] = "DEPassportNumber";
    KnownPiiCategory["DEIdentityCardNumber"] = "DEIdentityCardNumber";
    KnownPiiCategory["DETaxIdentificationNumber"] = "DETaxIdentificationNumber";
    KnownPiiCategory["DEValueAddedNumber"] = "DEValueAddedNumber";
    KnownPiiCategory["GRNationalIDCard"] = "GRNationalIDCard";
    KnownPiiCategory["GRNationalIDV2"] = "GRNationalIDV2";
    KnownPiiCategory["GRTaxIdentificationNumber"] = "GRTaxIdentificationNumber";
    KnownPiiCategory["HKIdentityCardNumber"] = "HKIdentityCardNumber";
    KnownPiiCategory["HUValueAddedNumber"] = "HUValueAddedNumber";
    KnownPiiCategory["HUPersonalIdentificationNumber"] = "HUPersonalIdentificationNumber";
    KnownPiiCategory["HUTaxIdentificationNumber"] = "HUTaxIdentificationNumber";
    KnownPiiCategory["INPermanentAccount"] = "INPermanentAccount";
    KnownPiiCategory["INUniqueIdentificationNumber"] = "INUniqueIdentificationNumber";
    KnownPiiCategory["IDIdentityCardNumber"] = "IDIdentityCardNumber";
    KnownPiiCategory["InternationalBankingAccountNumber"] = "InternationalBankingAccountNumber";
    KnownPiiCategory["IEPersonalPublicServiceNumber"] = "IEPersonalPublicServiceNumber";
    KnownPiiCategory["IEPersonalPublicServiceNumberV2"] = "IEPersonalPublicServiceNumberV2";
    KnownPiiCategory["ILBankAccountNumber"] = "ILBankAccountNumber";
    KnownPiiCategory["ILNationalID"] = "ILNationalID";
    KnownPiiCategory["ITDriversLicenseNumber"] = "ITDriversLicenseNumber";
    KnownPiiCategory["ITFiscalCode"] = "ITFiscalCode";
    KnownPiiCategory["ITValueAddedTaxNumber"] = "ITValueAddedTaxNumber";
    KnownPiiCategory["JPBankAccountNumber"] = "JPBankAccountNumber";
    KnownPiiCategory["JPDriversLicenseNumber"] = "JPDriversLicenseNumber";
    KnownPiiCategory["JPPassportNumber"] = "JPPassportNumber";
    KnownPiiCategory["JPResidentRegistrationNumber"] = "JPResidentRegistrationNumber";
    KnownPiiCategory["JPSocialInsuranceNumber"] = "JPSocialInsuranceNumber";
    KnownPiiCategory["JPMyNumberCorporate"] = "JPMyNumberCorporate";
    KnownPiiCategory["JPMyNumberPersonal"] = "JPMyNumberPersonal";
    KnownPiiCategory["JPResidenceCardNumber"] = "JPResidenceCardNumber";
    KnownPiiCategory["LVPersonalCode"] = "LVPersonalCode";
    KnownPiiCategory["LTPersonalCode"] = "LTPersonalCode";
    KnownPiiCategory["LUNationalIdentificationNumberNatural"] = "LUNationalIdentificationNumberNatural";
    KnownPiiCategory["LUNationalIdentificationNumberNonNatural"] = "LUNationalIdentificationNumberNonNatural";
    KnownPiiCategory["MYIdentityCardNumber"] = "MYIdentityCardNumber";
    KnownPiiCategory["MTIdentityCardNumber"] = "MTIdentityCardNumber";
    KnownPiiCategory["MTTaxIDNumber"] = "MTTaxIDNumber";
    KnownPiiCategory["NLCitizensServiceNumber"] = "NLCitizensServiceNumber";
    KnownPiiCategory["NLCitizensServiceNumberV2"] = "NLCitizensServiceNumberV2";
    KnownPiiCategory["NLTaxIdentificationNumber"] = "NLTaxIdentificationNumber";
    KnownPiiCategory["NLValueAddedTaxNumber"] = "NLValueAddedTaxNumber";
    KnownPiiCategory["NZBankAccountNumber"] = "NZBankAccountNumber";
    KnownPiiCategory["NZDriversLicenseNumber"] = "NZDriversLicenseNumber";
    KnownPiiCategory["NZInlandRevenueNumber"] = "NZInlandRevenueNumber";
    KnownPiiCategory["NZMinistryOfHealthNumber"] = "NZMinistryOfHealthNumber";
    KnownPiiCategory["NZSocialWelfareNumber"] = "NZSocialWelfareNumber";
    KnownPiiCategory["NOIdentityNumber"] = "NOIdentityNumber";
    KnownPiiCategory["PHUnifiedMultiPurposeIDNumber"] = "PHUnifiedMultiPurposeIDNumber";
    KnownPiiCategory["PLIdentityCard"] = "PLIdentityCard";
    KnownPiiCategory["PLNationalID"] = "PLNationalID";
    KnownPiiCategory["PLNationalIDV2"] = "PLNationalIDV2";
    KnownPiiCategory["PLPassportNumber"] = "PLPassportNumber";
    KnownPiiCategory["PLTaxIdentificationNumber"] = "PLTaxIdentificationNumber";
    KnownPiiCategory["PlregonNumber"] = "PLREGONNumber";
    KnownPiiCategory["PTCitizenCardNumber"] = "PTCitizenCardNumber";
    KnownPiiCategory["PTCitizenCardNumberV2"] = "PTCitizenCardNumberV2";
    KnownPiiCategory["PTTaxIdentificationNumber"] = "PTTaxIdentificationNumber";
    KnownPiiCategory["ROPersonalNumericalCode"] = "ROPersonalNumericalCode";
    KnownPiiCategory["RUPassportNumberDomestic"] = "RUPassportNumberDomestic";
    KnownPiiCategory["RUPassportNumberInternational"] = "RUPassportNumberInternational";
    KnownPiiCategory["SANationalID"] = "SANationalID";
    KnownPiiCategory["SGNationalRegistrationIdentityCardNumber"] = "SGNationalRegistrationIdentityCardNumber";
    KnownPiiCategory["SKPersonalNumber"] = "SKPersonalNumber";
    KnownPiiCategory["SITaxIdentificationNumber"] = "SITaxIdentificationNumber";
    KnownPiiCategory["SIUniqueMasterCitizenNumber"] = "SIUniqueMasterCitizenNumber";
    KnownPiiCategory["ZAIdentificationNumber"] = "ZAIdentificationNumber";
    KnownPiiCategory["KRResidentRegistrationNumber"] = "KRResidentRegistrationNumber";
    KnownPiiCategory["Esdni"] = "ESDNI";
    KnownPiiCategory["ESSocialSecurityNumber"] = "ESSocialSecurityNumber";
    KnownPiiCategory["ESTaxIdentificationNumber"] = "ESTaxIdentificationNumber";
    KnownPiiCategory["SQLServerConnectionString"] = "SQLServerConnectionString";
    KnownPiiCategory["SENationalID"] = "SENationalID";
    KnownPiiCategory["SENationalIDV2"] = "SENationalIDV2";
    KnownPiiCategory["SEPassportNumber"] = "SEPassportNumber";
    KnownPiiCategory["SETaxIdentificationNumber"] = "SETaxIdentificationNumber";
    KnownPiiCategory["SwiftCode"] = "SWIFTCode";
    KnownPiiCategory["CHSocialSecurityNumber"] = "CHSocialSecurityNumber";
    KnownPiiCategory["TWNationalID"] = "TWNationalID";
    KnownPiiCategory["TWPassportNumber"] = "TWPassportNumber";
    KnownPiiCategory["TWResidentCertificate"] = "TWResidentCertificate";
    KnownPiiCategory["THPopulationIdentificationCode"] = "THPopulationIdentificationCode";
    KnownPiiCategory["TRNationalIdentificationNumber"] = "TRNationalIdentificationNumber";
    KnownPiiCategory["UKDriversLicenseNumber"] = "UKDriversLicenseNumber";
    KnownPiiCategory["UKElectoralRollNumber"] = "UKElectoralRollNumber";
    KnownPiiCategory["UKNationalHealthNumber"] = "UKNationalHealthNumber";
    KnownPiiCategory["UKNationalInsuranceNumber"] = "UKNationalInsuranceNumber";
    KnownPiiCategory["UKUniqueTaxpayerNumber"] = "UKUniqueTaxpayerNumber";
    KnownPiiCategory["UsukPassportNumber"] = "USUKPassportNumber";
    KnownPiiCategory["USBankAccountNumber"] = "USBankAccountNumber";
    KnownPiiCategory["USDriversLicenseNumber"] = "USDriversLicenseNumber";
    KnownPiiCategory["USIndividualTaxpayerIdentification"] = "USIndividualTaxpayerIdentification";
    KnownPiiCategory["USSocialSecurityNumber"] = "USSocialSecurityNumber";
    KnownPiiCategory["UAPassportNumberDomestic"] = "UAPassportNumberDomestic";
    KnownPiiCategory["UAPassportNumberInternational"] = "UAPassportNumberInternational";
    KnownPiiCategory["Organization"] = "Organization";
    KnownPiiCategory["Email"] = "Email";
    KnownPiiCategory["URL"] = "URL";
    KnownPiiCategory["Age"] = "Age";
    KnownPiiCategory["PhoneNumber"] = "PhoneNumber";
    KnownPiiCategory["IPAddress"] = "IPAddress";
    KnownPiiCategory["Date"] = "Date";
    KnownPiiCategory["Person"] = "Person";
    KnownPiiCategory["Address"] = "Address";
    KnownPiiCategory["All"] = "All";
    KnownPiiCategory["Default"] = "Default";
})(KnownPiiCategory || (KnownPiiCategory = {}));
/** Known values of {@link ExtractiveSummarizationTaskParametersSortBy} that the service accepts. */
export var KnownExtractiveSummarizationTaskParametersSortBy;
(function (KnownExtractiveSummarizationTaskParametersSortBy) {
    KnownExtractiveSummarizationTaskParametersSortBy["Offset"] = "Offset";
    KnownExtractiveSummarizationTaskParametersSortBy["Rank"] = "Rank";
})(KnownExtractiveSummarizationTaskParametersSortBy || (KnownExtractiveSummarizationTaskParametersSortBy = {}));
/** Known values of {@link ErrorCodeValue} that the service accepts. */
export var KnownErrorCodeValue;
(function (KnownErrorCodeValue) {
    KnownErrorCodeValue["InvalidRequest"] = "InvalidRequest";
    KnownErrorCodeValue["InvalidArgument"] = "InvalidArgument";
    KnownErrorCodeValue["InternalServerError"] = "InternalServerError";
    KnownErrorCodeValue["ServiceUnavailable"] = "ServiceUnavailable";
    KnownErrorCodeValue["NotFound"] = "NotFound";
})(KnownErrorCodeValue || (KnownErrorCodeValue = {}));
/** Known values of {@link InnerErrorCodeValue} that the service accepts. */
export var KnownInnerErrorCodeValue;
(function (KnownInnerErrorCodeValue) {
    KnownInnerErrorCodeValue["InvalidParameterValue"] = "InvalidParameterValue";
    KnownInnerErrorCodeValue["InvalidRequestBodyFormat"] = "InvalidRequestBodyFormat";
    KnownInnerErrorCodeValue["EmptyRequest"] = "EmptyRequest";
    KnownInnerErrorCodeValue["MissingInputRecords"] = "MissingInputRecords";
    KnownInnerErrorCodeValue["InvalidDocument"] = "InvalidDocument";
    KnownInnerErrorCodeValue["ModelVersionIncorrect"] = "ModelVersionIncorrect";
    KnownInnerErrorCodeValue["InvalidDocumentBatch"] = "InvalidDocumentBatch";
    KnownInnerErrorCodeValue["UnsupportedLanguageCode"] = "UnsupportedLanguageCode";
    KnownInnerErrorCodeValue["InvalidCountryHint"] = "InvalidCountryHint";
})(KnownInnerErrorCodeValue || (KnownInnerErrorCodeValue = {}));
/** Known values of {@link WarningCode} that the service accepts. */
export var KnownWarningCode;
(function (KnownWarningCode) {
    KnownWarningCode["LongWordsInDocument"] = "LongWordsInDocument";
    KnownWarningCode["DocumentTruncated"] = "DocumentTruncated";
})(KnownWarningCode || (KnownWarningCode = {}));
/** Known values of {@link HealthcareEntityCategory} that the service accepts. */
export var KnownHealthcareEntityCategory;
(function (KnownHealthcareEntityCategory) {
    KnownHealthcareEntityCategory["BodyStructure"] = "BODY_STRUCTURE";
    KnownHealthcareEntityCategory["AGE"] = "AGE";
    KnownHealthcareEntityCategory["Gender"] = "GENDER";
    KnownHealthcareEntityCategory["ExaminationName"] = "EXAMINATION_NAME";
    KnownHealthcareEntityCategory["Date"] = "DATE";
    KnownHealthcareEntityCategory["Direction"] = "DIRECTION";
    KnownHealthcareEntityCategory["Frequency"] = "FREQUENCY";
    KnownHealthcareEntityCategory["MeasurementValue"] = "MEASUREMENT_VALUE";
    KnownHealthcareEntityCategory["MeasurementUnit"] = "MEASUREMENT_UNIT";
    KnownHealthcareEntityCategory["RelationalOperator"] = "RELATIONAL_OPERATOR";
    KnownHealthcareEntityCategory["Time"] = "TIME";
    KnownHealthcareEntityCategory["GeneORProtein"] = "GENE_OR_PROTEIN";
    KnownHealthcareEntityCategory["Variant"] = "VARIANT";
    KnownHealthcareEntityCategory["AdministrativeEvent"] = "ADMINISTRATIVE_EVENT";
    KnownHealthcareEntityCategory["CareEnvironment"] = "CARE_ENVIRONMENT";
    KnownHealthcareEntityCategory["HealthcareProfession"] = "HEALTHCARE_PROFESSION";
    KnownHealthcareEntityCategory["Diagnosis"] = "DIAGNOSIS";
    KnownHealthcareEntityCategory["SymptomORSign"] = "SYMPTOM_OR_SIGN";
    KnownHealthcareEntityCategory["ConditionQualifier"] = "CONDITION_QUALIFIER";
    KnownHealthcareEntityCategory["MedicationClass"] = "MEDICATION_CLASS";
    KnownHealthcareEntityCategory["MedicationName"] = "MEDICATION_NAME";
    KnownHealthcareEntityCategory["Dosage"] = "DOSAGE";
    KnownHealthcareEntityCategory["MedicationForm"] = "MEDICATION_FORM";
    KnownHealthcareEntityCategory["MedicationRoute"] = "MEDICATION_ROUTE";
    KnownHealthcareEntityCategory["FamilyRelation"] = "FAMILY_RELATION";
    KnownHealthcareEntityCategory["TreatmentName"] = "TREATMENT_NAME";
})(KnownHealthcareEntityCategory || (KnownHealthcareEntityCategory = {}));
/** Known values of {@link RelationType} that the service accepts. */
export var KnownRelationType;
(function (KnownRelationType) {
    KnownRelationType["Abbreviation"] = "Abbreviation";
    KnownRelationType["DirectionOfBodyStructure"] = "DirectionOfBodyStructure";
    KnownRelationType["DirectionOfCondition"] = "DirectionOfCondition";
    KnownRelationType["DirectionOfExamination"] = "DirectionOfExamination";
    KnownRelationType["DirectionOfTreatment"] = "DirectionOfTreatment";
    KnownRelationType["DosageOfMedication"] = "DosageOfMedication";
    KnownRelationType["FormOfMedication"] = "FormOfMedication";
    KnownRelationType["FrequencyOfMedication"] = "FrequencyOfMedication";
    KnownRelationType["FrequencyOfTreatment"] = "FrequencyOfTreatment";
    KnownRelationType["QualifierOfCondition"] = "QualifierOfCondition";
    KnownRelationType["RelationOfExamination"] = "RelationOfExamination";
    KnownRelationType["RouteOfMedication"] = "RouteOfMedication";
    KnownRelationType["TimeOfCondition"] = "TimeOfCondition";
    KnownRelationType["TimeOfEvent"] = "TimeOfEvent";
    KnownRelationType["TimeOfExamination"] = "TimeOfExamination";
    KnownRelationType["TimeOfMedication"] = "TimeOfMedication";
    KnownRelationType["TimeOfTreatment"] = "TimeOfTreatment";
    KnownRelationType["UnitOfCondition"] = "UnitOfCondition";
    KnownRelationType["UnitOfExamination"] = "UnitOfExamination";
    KnownRelationType["ValueOfCondition"] = "ValueOfCondition";
    KnownRelationType["ValueOfExamination"] = "ValueOfExamination";
})(KnownRelationType || (KnownRelationType = {}));
